source/extensions/filters/http/ext_proc/processor_state.h

```diff
@@ -49,14 +64,24 @@ class ProcessorState : public Logger::Loggable<Logger::Id::filter> {
   mutableBody(envoy::service::ext_proc::v3alpha::ProcessingRequest& request) const PURE;
 
 protected:
+  Filter* const parent_;
   Http::StreamFilterCallbacks* filter_callbacks_;
   CallbackState callback_state_ = CallbackState::Idle;
+  // Keep track of whether we must send the body when the callback is done
+  bool body_send_deferred_ = false;
+  // Keep track of whether we requested a watermark
+  bool watermark_requested_ = false;
   Http::HeaderMap* headers_ = nullptr;
   Event::TimerPtr message_timer_;
 };
 
 class DecodingProcessorState : public ProcessorState {
 public:
+  DecodingProcessorState(Filter* parent) : ProcessorState(parent) {}
+  DecodingProcessorState(const DecodingProcessorState&) = delete;
+  DecodingProcessorState& operator=(const DecodingProcessorState&) = delete;
+  ~DecodingProcessorState() override;
```

Comment:
I think you can omit these if you're just declaring them as `= default`

Reply:
I don't know why I ran in to problems before, but with the destructors deleted it works fine.